server/controllers/card.js

```diff
@@ -0,0 +1,241 @@
+  if (cardItem === "attachment") {
+    const imageObject = {
+      imageName: req.files[0].originalname,
+      imageUrl: req.files[0].path,
+      imageId: "",
+    };
+    const uploadStatus = await cloud.uploads(imageObject.imageUrl);
+    itemContent = {
+      imageName: req.files[0].originalname,
+      imageUrl: uploadStatus.url,
+      imageId: uploadStatus.id,
+    };
+
+    await unlinkFile(req.files[0].path);
+  }
+
+  const targetItem = `boards.$[board].columns.$[column].cards.$[card].${cardItem}`;
+  const updateStatus = await User.updateOne(
+    { _id: userId, "boards.columns.cards._id": cardObjectId },
+    {
+      $set: {
+        [targetItem]: itemContent,
+      },
+    },
+    {
+      arrayFilters: [
+        { "board._id": boardObjectId },
+        { "column._id": columnObjectId },
+        { "card._id": cardObjectId },
+      ],
+    }
+  );
+
+  if (updateStatus.nModified === 1) {
+    return res.status(200).json({ success: true });
+  }
+
+  res.status(500);
+  throw new Error("Something went wrong");
+});
+
+exports.removeCardItems = asyncHandler(async (req, res, next) => {
+  const { cardItem, userId, cardId } = req.body;
+
+  const cardObjectId = ObjectID(cardId);
+  const getDocumentIndex = (document) => {
+    for (const boardIndex in document[0].boards) {
+      for (const columnIndex in document[0].boards[boardIndex].columns) {
+        for (const cardIndex in document[0].boards[boardIndex].columns[
+          columnIndex
+        ].cards) {
+          const documentCardId =
+            document[0].boards[boardIndex].columns[columnIndex].cards[cardIndex]
+              ._id;
+          if (documentCardId.toString() === cardObjectId.toString()) {
+            return {
+              boardIndex,
+              columnIndex,
+              cardIndex,
+            };
+          }
+        }
+      }
+    }
+  };
+
+  const document = await User.find({
+    _id: userId,
+    "boards.columns.cards._id": cardObjectId,
+  });
+  const { boardIndex, columnIndex, cardIndex } = getDocumentIndex(document);
+
+  const targetItem = `boards.${boardIndex}.columns.${columnIndex}.cards.${cardIndex}.${cardItem}`;
+
+  const removeStatus = await User.updateOne(
+    { _id: userId, "boards.columns.cards._id": cardObjectId },
+    {
+      $unset: { [targetItem]: "" },
+    }
+  );
+  if (removeStatus.nModified === 1) {
+    return res.status(200).json({ success: true });
+  }
+  res.status(500);
+  throw new Error("Something went wrong");
+});
+
+exports.createChecklist = asyncHandler(async (req, res, next) => {
+  const { checklistItem, cardId, columnId, boardId, userId } = req.body;
+  const columnObjectId = ObjectID(columnId);
+  const boardObjectId = ObjectID(boardId);
+  const cardObjectId = ObjectID(cardId);
+  const createStatus = await User.updateOne(
+    { _id: userId, "boards.columns.cards._id": cardObjectId },
+    {
+      $push: {
+        "boards.$[board].columns.$[column].cards.$[card].checklists": {
```

Comment:
As mentioned many times, this is actually not needed. We are doing a different approach so we can remove this code. Please make sure to be aware of what we're doing in the project going forward so we aren't spending time on unnecessary changes 👍🏻